virtcontainers/clh.go

```diff
@@ -0,0 +1,1260 @@
+		consoleType: cctOFF,
+	})
+
+	return nil
+}
+
+func (clh *cloudHypervisor) startSandbox(timeout int) error {
+	span, _ := clh.trace("startSandbox")
+	defer span.Finish()
+
+	clh.Logger().WithField("function", "startSandbox").Info("starting Sandbox")
+
+	vmPath := filepath.Join(store.RunVMStoragePath(), clh.id)
+	err := os.MkdirAll(vmPath, store.DirMode)
+	if err != nil {
+		return err
+	}
+
+	if clh.config.SharedFS == config.VirtioFS {
+		clh.Logger().WithField("function", "startSandbox").Info("Starting virtiofsd")
+		timeout, err = clh.setupVirtiofsd(timeout)
+		if err != nil {
+			return err
+		}
+		if err = clh.storeState(); err != nil {
+			return err
+		}
+	}
+
+	var strErr string
+	var pid = 0
+	strErr, err, pid = clh.LaunchIch()
+	if err != nil {
+		return fmt.Errorf("failed to launch cloud-hypervisor: %s, error messages from log: %s", err, strErr)
+	}
+	if err := clh.waitVMM(clhTimeout); err != nil {
+		clh.Logger().WithField("error", err).Warn("cloud-hypervisor init failed")
+		clh.shutdownVirtiofsd()
+		return err
+	}
+
+	clh.state.PID = pid
+	clh.state.state = clhReady
+	clh.storeState()
+
+	return nil
+}
+
+func (clh *cloudHypervisor) getSandboxConsole(id string) (string, error) {
+	clh.Logger().WithField("function", "getSandboxConsole").WithField("id", id).Info("Get Sandbox Console")
+	return "", nil
+}
+
+func (clh *cloudHypervisor) disconnect() {
+	clh.Logger().WithField("function", "disconnect").Info("Disconnecting Sandbox Console")
+}
+
+func (clh *cloudHypervisor) getThreadIDs() (vcpuThreadIDs, error) {
+
+	clh.Logger().WithField("function", "getThreadIDs").Info("get thread ID's")
+
+	var vcpuInfo vcpuThreadIDs
+
+	vcpuInfo.vcpus = make(map[int]int)
+
+	return vcpuInfo, nil
+}
+
+func (clh *cloudHypervisor) hotplugAddDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
+	clh.Logger().WithField("function", "hotplugAddDevice").Info("Add hotplug device")
+	return nil, nil
+}
+
+func (clh *cloudHypervisor) hotplugRemoveDevice(devInfo interface{}, devType deviceType) (interface{}, error) {
+	clh.Logger().WithField("function", "hotplugRemoveDevice").Info("Remove hotplug device")
+	return nil, nil
+}
+
+func (clh *cloudHypervisor) hypervisorConfig() HypervisorConfig {
+	clh.Logger().WithField("function", "hypervisorConfig").Info("get hypervisor config")
+	return clh.config
+}
+
+func (clh *cloudHypervisor) resizeMemory(reqMemMB uint32, memoryBlockSizeMB uint32, probe bool) (uint32, memoryDevice, error) {
+	clh.Logger().WithFields(log.Fields{
+		"function":       "resizeMemory",
+		"reqMemMB":       reqMemMB,
+		"memBlockSizeMB": memoryBlockSizeMB,
+	}).Info("resize the VCPU's ")
+	return 0, memoryDevice{}, nil
+}
+
+func (clh *cloudHypervisor) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint32, err error) {
+
+	clh.Logger().WithFields(log.Fields{
+		"function": "resizeVCPUs",
+		"curr":     currentVCPUs,
+		"new":      newVCPUs,
+	}).Info("resize the VCPU's ")
```

Comment:
Ditto

Reply:
We seem to have lost the explanation for what the ditto means here @sboeuf ;)